pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BaseTableUpsertMetadataManager.java

```diff
@@ -61,11 +62,12 @@ public void init(TableConfig tableConfig, Schema schema, TableDataManager tableD
     PartialUpsertHandler partialUpsertHandler = null;
     if (upsertConfig.getMode() == UpsertConfig.Mode.PARTIAL) {
       Map<String, UpsertConfig.Strategy> partialUpsertStrategies = upsertConfig.getPartialUpsertStrategies();
-      Preconditions.checkArgument(partialUpsertStrategies != null,
+      String rowMergerCustomImplementation = upsertConfig.getPartialUpsertMergerClass();
+      Preconditions.checkArgument(
+          StringUtils.isNotBlank(rowMergerCustomImplementation) || partialUpsertStrategies != null,
```

Comment:
(minor) These checks can be pushed down to the constructor of `PartialUpsertHandler`

Reply:
If the checks are moved down to `PartialUpsertHandler`, the create table API response is success but the table goes in error state. This check would pro-actively fail table creation.

Reply:
I've applied a commit to demonstrate the idea. The exception will be thrown from the constructor

Reply:
Got it, thank you Jackie.